### PR TITLE
Add Python3 compatibility

### PR DESCRIPTION
**Pull request by dadrian**
This PR allows zschema to function under Python 2 and Python 3. It updates CI to run tests on both versions.

See individual commit messages (although the Python 3 change is pretty big, the message is more descriptive)

### CODE REVIEW

File: zschema/keys.py [hunk 1]
```diff
@@ -21,34 +26,42 @@ def __init__(self, port):
...
 
     def __eq__(self, other):
-        if type(other) == int:
-            return int(self.port).__eq__(other)
-        elif type(other) in (str, unicode):
-            return self.port.__eq__(str(other))
+        if isinstance(other, int):
+            return int(self.port) == other
+        elif isinstance(other, string_types):
+            return self.port.__eq__(other)
```
Comment (zakird) [inline zschema/keys.py]: this feels inconsistent with L41

Comment (dadrian) [reply zschema/keys.py]: int doesn't have `__eq__` in Python 3, but may as well make all these `==`

**Comment by zakird** [comment]
This looks solid.

File: zschema/keys.py [hunk 2]
```diff
@@ -21,34 +26,42 @@ def __init__(self, port):
...
 
-    def __cmp__(self, other):
-        if type(other) == int:
-            return cmp(int(self.port), other)
-        elif type(other) in (str, unicode):
-            return cmp(self.port, str(other))
+    def __lt__(self, other):
+        if isinstance(other, int):
+            return self.port < str(other)
+        elif isinstance(other, string_types):
```
Comment (justinbastress) [inline zschema/keys.py]: I realize this keeps the old behavior...but do we really want to sort port numbers as strings?

Comment (dadrian) [reply zschema/keys.py]: Probably not, but I didn't want to make that change here. Also, if we compare them as integers it gets complicated if we're comparing them against a non-integer string, such as `"example"`

**Pull request merged**